crates/core/src/syn/parser/stmt/mod.rs

```diff
@@ -217,7 +218,18 @@ impl Parser<'_> {
 			_ => {
 				// TODO: Provide information about keywords.
 				let value = ctx.run(|ctx| self.parse_value_field(ctx)).await?;
-				Ok(Self::refine_stmt_value(value))
+				if let SqlValue::Expression(x) = &value {
+					if let Expression::Binary {
+						l: SqlValue::Param(ref x),
+						o: Operator::Equal,
+						..
+					} = **x
+					{
+						let span = token.span.covers(self.recent_span());
+						bail!("Variable declaration without `let` is deprecated", @span => "replace with `let {x} = ..`")
```

Comment:
In fact, that's not anymore a depreceation, it is an error.

Shall we just say: 

```suggestion
						bail!("Variable declaration must start with `let`", @span => "replace with `let {x} = ..`")
```

Reply:
I wanted to make it a deprecation with an error so that in the next release we can make this be a equality test.

Reply:
The deprecation makes it more clear that the behavior around this syntax might change in the future.